libs/vkd3d/command.c

```diff
@@ -10214,6 +10307,41 @@ static void d3d12_command_list_execute_query_resolve(struct d3d12_command_list *
     }
 }
 
+static void d3d12_command_list_flush_query_resolves(struct d3d12_command_list *list)
+{
+    unsigned int i;
+
+    if (!list->query_resolve_count)
+        return;
+
+    d3d12_command_list_update_descriptor_buffers(list);
+
+    for (i = 0; i < list->query_resolve_count; i++)
+        d3d12_command_list_execute_query_resolve(list, &list->query_resolves[i]);
+
+    list->query_resolve_count = 0;
+
+    hash_map_clear(&list->query_resolve_lut);
```

Comment:
We might want a way to avoid freeing the memory and allocate it over and over. hash_map_reset() perhaps?

Reply:
Done, `clear` now only clears while `free` deallocates memory.